src/screen/ShopScreen.java

```diff
@@ -0,0 +1,203 @@
+        super.update();
+
+        draw();
+        if (this.selectionCooldown.checkFinished()
+                && this.inputDelay.checkFinished()
+                && this.money_alertCooldown.checkFinished()
+                && this.max_alertCooldown.checkFinished()) {
+
+            if (inputManager.isKeyDown(KeyEvent.VK_UP)
+                    || inputManager.isKeyDown(KeyEvent.VK_W)) {
+                previousMenuItem();
+                this.selectionCooldown.reset();
+            }
+            if (inputManager.isKeyDown(KeyEvent.VK_DOWN)
+                    || inputManager.isKeyDown(KeyEvent.VK_S)) {
+                nextMenuItem();
+                this.selectionCooldown.reset();
+            }
+            if (inputManager.isKeyDown(KeyEvent.VK_SPACE))
+            {
+                switch (selected_item) {
+                    case 1:
+                        if (usecoinforupgrade(wallet.getBullet_lv())) {
+                            wallet.setBullet_lv(wallet.getBullet_lv() + 1);
+                            this.selectionCooldown.reset();
+                        }
+                        break;
+                    case 2:
+                        if (usecoinforupgrade(wallet.getShot_lv())) {
+                            wallet.setShot_lv(wallet.getShot_lv() + 1);
+                            this.selectionCooldown.reset();
+                        }
+                        break;
+                    case 3:
+                        if (usecoinforupgrade(wallet.getLives_lv())) {
+                            wallet.setLives_lv(wallet.getLives_lv() + 1);
+                            this.selectionCooldown.reset();
+                        }
+                        break;
+                    case 4:
+                        if (usecoinforupgrade(wallet.getCoin_lv())) {
+                            wallet.setCoin_lv(wallet.getCoin_lv() + 1);
+                            this.selectionCooldown.reset();
+                        }
+                        break;
+                    default:
+                        break;
+                }
+            }
+
+            if (inputManager.isKeyDown(KeyEvent.VK_ESCAPE))
+                this.isRunning = false;
+        }
+    }
+
+    /**
+     * Shifts the focus to the next shop item.
+     */
+    private void nextMenuItem() {
+        if (this.selected_item == 4)
+            this.selected_item = 1;
+        else
+            this.selected_item++;
+    }
+
+    /**
+     * Shifts the focus to the previous shop item.
+     */
+    private void previousMenuItem() {
+        if (this.selected_item == 1)
+            this.selected_item = 4;
+        else
+            this.selected_item--;
+    }
+
+    private void draw() {
+        drawManager.initDrawing(this);
+
+
+        drawManager.drawShop(this,selected_item,wallet,money_alertCooldown,max_alertCooldown);
+
+        drawManager.completeDrawing(this);
+    }
+
+    public boolean usecoinforupgrade(int level)
```

Comment:
Please use java naming convention (camelCase).
And I think this method name is just `upgrade`
```java
public boolean upgrade(int level)
```

Reply:
Renamed it.